main.go

```diff
@@ -105,10 +127,24 @@ func getIntEnv(envVar string) int {
 	return i
 }
 
+func getNotifier() (notify.Notifier, error) {
+	switch notifierType {
+	case TelegramNotifierType:
+		return notify.NewTelegram(tgApiToken, tgUsername)
+	default:
+		return notify.NewEmail(email, password), nil
+	}
+}
+
 func Run(args []string) error {
 	if err := checkFlags(); err != nil {
 		return err
 	}
+	var err error
+	notifier, err = getNotifier()
```

Comment:
```suggestion
	notifier, err := getNotifier()
```
pass `notifier` as an arg ro checkSlots()

Reply:
yeah